doc/dev/radosgw/virtual_hosting.md

```diff
@@ -0,0 +1,122 @@
+# radosgw virtual hosting
+
+allow the different rest apis to use their own domain names. route requests between them based on Host header
+
+## motivation
+
+radosgw currently serves all rest apis from the same namespace. some are hosted at the root, with others under paths like `admin`, `auth`, `swift` etc. most apis are enabled by default, so parts of the root s3 namespace are unavailable because those paths get routed to other apis
+
+s3 shares the root path with several other apis (iam, s3control, sts, sns) and relies on tricks to disambiguate the requests and route them correctly. these tricks are based on assumptions that may not be reliable, especially as the apis evolve
+
+to avoid these conflicts in path-based routing, we can first route requests by hostname
```

Comment:
Is this a suggestion for something along the lines of Apache's VirtualHost?

Reply:
S3
APIs
These
s/tricks/heuristics/

To avoid